### PR TITLE
Add authentication for organizations

### LINKED ISSUE #15: Add authentication for organizations
Users will login as their organization and will have permissions to do everything.

Use JWT for communication.

This library can handle JWT communicate for Angular 2: https://github.com/auth0/angular2-jwt

### PR DESCRIPTION
**Pull request by emmanuelroussel**
Closes #15 

Login now sends the credentials to the api to authenticate. It stores the token in storage and the `angular2-jwt` library will automatically add it in the auth header for every request to the api. The call inside login uses `http` instead of angular2-jwt's `authHttp` because this is the only call where we don't want a header.

Every time a user opens the app, we verify if the token is expired. If not, we go to the TabsPage, if yes, we go to LoginPage.

I also added a logout button in a left side menu. Eventually, we might want to put some info about the user and organization in this menu.

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -16,6 +19,14 @@ import { InventoryData } from '../providers/inventory-data';
 import { StockpileData } from '../providers/stockpile-data';
 import { UserData } from '../providers/user-data';
 
+let storage = new Storage();
+
+export function getAuthHttp(http) {
+  return new AuthHttp(new AuthConfig({
+    tokenGetter: (() => storage.get('id_token')),
```
Comment (AdamVig) [inline src/app/app.module.ts]: Why use snake case here for `id_token`? Shouldn't it be `idToken`?

(Used in a few places in the user data provider as well.)

File: src/providers/user-data.ts [hunk 2]
```diff
@@ -1,11 +1,54 @@
...
 
   login(email: string, password: string) {
-    return Promise.resolve();
+    let creds = {
+      'email': email,
+      'password': password
+    };
+
+    return new Promise((resolve, reject) => {
+      this.http.post(this.stockpileData.getUrl(Links.authenticate), creds)
```
Comment (AdamVig) [inline src/providers/user-data.ts]: This line of code is great. Shows that the HAL method is actually doing what it promises: keeping hardcoded API URLs out of our code.

File: src/app/app.module.ts (hunk 1 above)
Comment (emmanuelroussel) [inline src/app/app.module.ts]: I'm not sure why, but it seems to be the standard to name the auth token `id_token`. It is that name default, it simply has to be overwritten to use ionic storage.

File: src/providers/user-data.ts (hunk 2 above)
Comment (emmanuelroussel) [inline src/providers/user-data.ts]: I'm actually not sure I'm using HAL the right way here. There is no hard coded urls, but we need hard coded strings to get the urls (they are in `constants.ts`). It does seem like a slight improvement (urls can be changed more easily), but still feels odd to do this.

**Pull request merged**